libr/util/print.c

```diff
@@ -1581,9 +1582,57 @@ R_API void r_print_progressbar(RPrint *p, int pc, int _cols) {
 	p->cb_printf ("]");
 }
 
+/* TODO: handle screen width */
+R_API void r_print_progressbar_with_count(RPrint *p, unsigned int pc, unsigned int total, int _cols, bool reset_line) {
+	int i, cols = (_cols == -1)? 78: _cols;
+	if (!p) {
+		p = &staticp;
+	}
+	const bool enable_colors = p && (p->flags & R_PRINT_FLAGS_COLOR);
+	const char *h_line = p->cons->use_utf8? RUNE_LONG_LINE_HORIZ: "-";
+	const char *block = p->cons->use_utf8? UTF_BLOCK: "#";
+
+	total = R_MAX (1, total);
+	pc = R_MAX (0, R_MIN (total, pc));
+	if (reset_line) {
+		p->cb_printf ("\r");
+	}
+	if (p->flags & R_PRINT_FLAGS_HEADER) {
+		if (enable_colors) {
+			p->cb_printf ("%s%4d%s%% %s%6d%s/%6d ", Color_GREEN, pc * 100 / total, Color_RESET, Color_GREEN, pc, Color_RESET, total, Color_YELLOW);
+		} else {
+			p->cb_printf ("%4d%% %6d/%6d ", pc * 100 / total, pc, total);
+		}
+		// TODO: determine string length of the numbers
+		cols -= 20;
+	}
+	if (cols > 0) {
+		if (enable_colors) {
+			p->cb_printf ("[%s", Color_YELLOW);
```

Comment:
just oneline with ternary operator enable_colors?Color_YELLOW:""

also, instead of hardcoding the yellow and green. you may take the colors from the selected palette

Reply:
How do I grab the colors from the palette?

Reply:
Use `r_cons_pal_get_i`, you can resolve the types by name with `r_cons_pal_get_name`. see cons/pal.c for all the possible values. The defines + enum in r_cons.h must be removed